jobclient/python/instance.py

```diff
@@ -0,0 +1,208 @@
+                 task_id: UUID,
+                 slave_id: str,
+                 executor_id: str,
+                 start_time: int,
+                 hostname: str,
+                 status: Status,
+                 preempted: bool,
+                 # Optional arguments
+                 end_time: Optional[int] = None,
+                 progress: Optional[int] = None,
+                 progress_message: Optional[str] = None,
+                 reason_code: Optional[int] = None,
+                 output_url: Optional[str] = None,
+                 executor: Optional[Executor] = None,
+                 reason_mea_culpa: Optional[bool] = None):
+        """Initialize an instance.
+
+        Required Parameters
+        -------------------
+        :param slave_id: The ID of the slave running this instance.
+        :type slave_id: str
+        :param executor_id: The ID of the executor running this instance.
+        :type executor_id: str
+        :param start_time: The UNIX timestamp at which this instance began
+            running.
+        :type start_time: int
+        :param hostname: The hostname of the machine running this instance.
+        :type hostname: str
+        :param status: The status of this instance.
+        :type status: Status
+        :param preempted: If true, then this instance was preempted.
+        :type preempted: bool
+        Optional Parameters
+        -------------------
+        :param progress: Progress of this instance.
+        :type progress: Optional[int], optional
+        :param progress_message: Progress message of this instance.
+        :type progress_message: Optional[str], optional
+        :param reason_code: Reason code of this instance.
+        :type reason_code: Optional[int], optional
+        :param output_url: Output URL for this instance.
+        :type output_url: Optional[str], optional
+        :param executor: Executor information for this instance.
+        :type executor: Optional[Executor], optional
+        :param reason_mea_culpa: If true, then reason mea culpa.
+        :type reason_mea_culpa: Optional[bool], optional
+        """
+        self.task_id = task_id
+        self.slave_id = slave_id
+        self.executor_id = executor_id
+        self.start_time = start_time
+        self.hostname = hostname
+        self.status = status
+        self.preempted = preempted
+        self.end_time = end_time
+        self.progress = progress
+        self.progress_message = progress_message
+        self.reason_code = reason_code
+        self.output_url = output_url
+        self.executor = executor
+        self.reason_mea_culpa = reason_mea_culpa
+
+    def __hash__(self):
+        PRIME = 31
+        result = 1
+        result = (PRIME * result +
+                  hash(self.status) if self.status is not None else 0)
+        result = (PRIME * result +
+                  hash(self.task_id) if self.task_id is not None else 0)
+        return result
```

Comment:
What is this used for?

Reply:
It was in the original Java client code, but from a quick glance of it it's not actively used anywhere.